src/Arriba/Arriba.Composition/Composition/Host.cs

```diff
@@ -3,93 +3,94 @@
+            services.AddSingleton<IObjectCacheFactory, MemoryCacheFactory>();
+            services.AddSingleton<SecureDatabase>();
+            services.AddSingleton<DatabaseFactory>();
+            services.AddSingleton<ICorrector, TodayCorrector>();
+
+            services.AddTransient<IRoutedApplication, ArribaImportApplication>();
+            services.AddTransient<IRoutedApplication, ArribaQueryApplication>();
+            services.AddTransient<IRoutedApplication, ArribaTableRoutesApplication>();
+
+            services.AddTransient<IApplication, RoutedApplicationHandler>();
+            services.AddSingleton<ApplicationServer, ComposedApplicationServer>();
+        }
 
-            conventions.ForTypesDerivedFrom<IContentWriter>()
-                .Export()
-                .Export<IContentWriter>()
-                .Shared();
+        public static void AddDerrivedTypes<T>(this IServiceCollection services, IEnumerable<Type> types, ServiceLifetime lifetime = ServiceLifetime.Transient)
+        {
+            var serviceType = typeof(T);
+            var query = types
+                .Where(instanceType => !instanceType.IsInterface && !instanceType.IsAbstract)
+                .Where(instanceType => serviceType.IsAssignableFrom(instanceType));
 
-            conventions.ForTypesDerivedFrom<IApplication>()
-                .ExportInterfaces()
-                .Shared();
+            foreach (Type t in query)
+            {
+                services.Add(new ServiceDescriptor(serviceType, t, lifetime));
+            }
+        }
 
+        private static IEnumerable<Type> GetArribaTypes()
+        {
             var assemblies = new[] {
                 // Arriba.dll
                 typeof(Table).Assembly, 
-                // Arriba.Composition.dll
-                typeof(Host).Assembly,
                 // Arriba.Adapter.Newtonsoft
                 typeof(JsonContentWriter).Assembly
             };
 
-            _configuration = new ContainerConfiguration().WithAssemblies(assemblies.Distinct(), conventions);
-            Compose();
-        }
-
-        public void Compose()
-        {
-            _container = _configuration.CreateContainer();
-        }
-
-        public void Add<TContract>(TContract value)
-        {
-            _configuration.WithExport<TContract>(value);
-        }
-
-        public void AddConfigurationValue<T>(string name, T value)
-        {
-            _configuration.WithExport<T>(value, contractName: name);
-        }
-
-        public T GetService<T>()
-        {
-            return _container.GetExport<T>();
-        }
-
-        public IEnumerable<T> GetServices<T>()
-        {
-            return _container.GetExports<T>();
+            return assemblies.SelectMany(x => x.GetTypes())
+                .ToList();
         }
-
-        public void Dispose()
+    }
+    
+    public class ArribaServiceProvider
+    {
+        public static IServiceProvider Create()
```

Comment:
I'm in doubt if it would be better to have the DebugSecurityConfig  passed  by default at the instance creation or registering a service when is needed specifying the SecurityConfiguration properties when needed. 

I think that it better to have a ISecurityConfiguration parameter on Create and use the DebugSecurityConfig when it is null.

Reply:
I moved this into the Test assembly since it's only required for testing at this point and it was a little confusing when looking back at it. The change sounds relevant, however not yet demanded by the code so I'll refrain from making it.